src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -151,6 +151,12 @@ static void HandleNodeBrowse(void * context, DnssdService * services, size_t ser
 {
     ResolverDelegateProxy * proxy = static_cast<ResolverDelegateProxy *>(context);
 
+    if (error != CHIP_NO_ERROR && error != CHIP_END_OF_INPUT)
```

Comment:
END_OF_INPUT does not seem to be an error. Could we design our logic in some other way so that `error` actually is used as an error?